common_types/src/schema.rs

```diff
@@ -595,6 +630,16 @@ impl Schema {
         self.timestamp_index
     }
 
+    /// Whether i-nth column is key column
+    /// TODO: should tag column be key columns?
+    pub fn non_key_column(&self, i: usize) -> bool {
```

Comment:
Maybe `is_key_column` is more suitable for this kind of predictor.

And maybe we need to also explain what "key column" is in the comment